app/src/main/java/com/edulectronics/tinycircuit/ui/adapters/CircuitAdapter.java

```diff
@@ -56,12 +57,15 @@ public View getView(int position, View view, ViewGroup parent) {
         LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
         View row = inflater.inflate(R.layout.component, parent, false);
         ImageView image = (ImageView) row.findViewById(R.id.imageView);
-        image.setImageResource(getImageId(position));
-        return row;
-    }
 
-    private int getImageId(int position) {
-        // TODO: replace with components image
-        return R.mipmap.ic_launcher;
+        ImageFactory imageFactory = new ImageFactory();
+        Object object = getItem(position);
+        if(object != null){
+            image.setImageResource(imageFactory.GetImage(object));
```

Comment:
Misschien is het in android inderdaad toch makkelijker om het component model de referentie naar de image bij te laten houden (als in component.getImage()). Scheelt een hoop if statements. Als we dan meerdere stylen willen implementeren (bijv. schematisch), dan kunnen we hem gewoon aanpassen naar component.getImage(currentMode)